fs/fs.go

```diff
@@ -29,7 +29,7 @@ func NewDatastore(path string) (ds.Datastore, error) {
 
 // KeyFilename returns the filename associated with `key`
 func (d *Datastore) KeyFilename(key ds.Key) string {
-	return filepath.Join(d.path, key.String(), ObjectKeySuffix)
```

Comment:
The flat version would still be just a Datastore, it'd just use the filesystem differently than this package. S3 plans are not affected by that.

Summarizing: we leave this package as is, with some prominent caveats that it should only be used with human-friendly, trusted keys, and I make a new package alongside. That sounds good.